edb/server/compiler/compiler.py

```diff
@@ -537,6 +537,71 @@ def _compile_ql_script(
 
         return sql.decode(), argmap
 
+    def _compile_query_args(
+            self,
+            ctx: CompileContext,
+            schema: s_schema.Schema,
+            subtypes: List[Tuple[str, Any, bool]],
+    ) -> Tuple[bytes, uuid.UUID]:
+        # TODO(fmoor) perhaps check for zero parameters and
+        # return a static value (b'...', uuid.UUID(b'...'))
+        if not subtypes:
+            return b'\x01(\xda\xba\tJ\x8b\x86h\xf8!T\xb8]\xf6\xe6\x0e\x00\x00', uuid.UUID('28daba09-4a8b-8668-f821-54b85df6e60e')
```

Comment:
@elprans is something like this valid? I have no idea if the descriptor id is stable or not. Skipping compilation of zero arguments side steps a class of errors where `qlcompiler.compile_ast_to_ir()` raises an exception about some type already existing that only happens when the AST being compiled is ~ `SELECT {};`.

Reply:
What is the error you're seeing?  If manually constructed `SELECT {}` (as in shape, not empty set), doesn't work then `SELECT std::FreeObject` should surely work.  I'd very much like to avoid hardcoding serialized descriptors like this.